symengine/functions.cpp

```diff
@@ -3339,9 +3344,18 @@ RCP<const Basic> abs(const RCP<const Basic> &arg)
         } else {
             return arg_;
         }
+    } else if (is_a<Complex>(*arg)) {
+        RCP<const Complex> arg_ = rcp_static_cast<const Complex>(arg);
+        RCP<const Complex> conjugate = rcp_static_cast<const Complex>(sub((*arg_).real_part(),mul(I,(*arg_).imaginary_part())));
+        return sqrt(abs(((*arg_).mulcomp(*conjugate))));
```

Comment:
Writing `sqrt(real*real+imaginary*imaginary)` in terms of `mul()`, `real_part()` etc. also led to a long statement. Also I felt that the return statement looked clearer in the present case.

`abs` call removed. :+1: